### PR TITLE
Add virtio over ivshmem for acrn and QNX.

### PR DESCRIPTION
**Pull request by YuanyuanZhao609**
EB was tested, QNX and android works normally.

Tracked-On: OAM-119803

### CODE REVIEW

File: groups/kernel/gmin64/config-lts/linux-intel-lts2022/x86_64_defconfig [hunk 1]
```diff
@@ -7747,7 +7746,30 @@ CONFIG_RUNTIME_TESTING_MENU=y
...
+CONFIG_QNX_GUEST=y
+CONFIG_ACPI_HOTPLUG_MEMORY=y
+CONFIG_ARCH_MEMORY_PROBE=y
+CONFIG_DEVICE_PRIVATE=y
+CONFIG_FS_DAX=n
+CONFIG_MEMORY_HOTPLUG_DEFAULT_ONLINE=n
+CONFIG_PCI_P2PDMA=n
+CONFIG_VIRTIO_IVSHMEM=y
+CONFIG_VIRTIO_IVSHMEM_IVSHMEM=y
+CONFIG_VIRTIO_IVSHMEM_GUEST_SHM=y
```
Comment (xyzhao2018) [inline groups/kernel/gmin64/config-lts/linux-intel-lts2022/x86_64_defconfig:7763]: Please move virtualization related config to virt_diffconfig.

File: groups/kernel/gmin64/config-lts/linux-intel-lts2022/x86_64_defconfig [hunk 2]
```diff
@@ -7747,7 +7746,30 @@ CONFIG_RUNTIME_TESTING_MENU=y
 CONFIG_ARCH_USE_MEMTEST=y
 # CONFIG_MEMTEST is not set
 # end of Kernel Testing and Coverage
-
+CONFIG_IVSHM=y
+CONFIG_IVSHM_IVSHMEM=y
+CONFIG_IVSHM_GUEST_SHM=y
+CONFIG_MEMORY_HOTPLUG=y
```
Comment (bhe4) [inline groups/kernel/gmin64/config-lts/linux-intel-lts2022/x86_64_defconfig:7752]: I checked some configurations are already exist in virt_diffconfig
CONFIG_IVSHM=y
CONFIG_IVSHM_IVSHMEM=y
CONFIG_IVSHM_GUEST_SHM=y

File: groups/kernel/gmin64/config-lts/linux-intel-lts2022/x86_64_defconfig [hunk 3]
```diff
@@ -440,7 +440,7 @@ CONFIG_X86_NEED_RELOCS=y
 CONFIG_PHYSICAL_ALIGN=0x1000000
 CONFIG_DYNAMIC_MEMORY_LAYOUT=y
 CONFIG_RANDOMIZE_MEMORY=y
-CONFIG_RANDOMIZE_MEMORY_PHYSICAL_PADDING=0x0
+CONFIG_RANDOMIZE_MEMORY_PHYSICAL_PADDING=0xa
```
Comment (bhe4) [inline groups/kernel/gmin64/config-lts/linux-intel-lts2022/x86_64_defconfig:443]: what's the purpose to set it as 0xa?

**Review by bhe4 — CHANGES_REQUESTED** [review]

File: groups/kernel/gmin64/config-lts/linux-intel-lts2022/x86_64_defconfig (hunk 2 above)
Comment (YuanyuanZhao609) [reply groups/kernel/gmin64/config-lts/linux-intel-lts2022/x86_64_defconfig:7752]: OK. Moved.

File: groups/kernel/gmin64/config-lts/linux-intel-lts2022/virt_diffconfig [hunk 4]
```diff
@@ -30,3 +30,12 @@ CONFIG_GPIO_VIRTIO=y
 CONFIG_I2C_VIRTIO=y
 CONFIG_ACRN_GUEST=y
 CONFIG_SPI_VIRTIO=y
+CONFIG_VIRTIO_IVSHMEM=y
+CONFIG_VIRTIO_IVSHMEM_IVSHMEM=y
+CONFIG_VIRTIO_IVSHMEM_GUEST_SHM=y
+CONFIG_VIRTIO_IVSHMEM_DEBUG=n
+CONFIG_VIRTIO_MEM=n
```
Comment (JeevakaPrabu) [inline groups/kernel/gmin64/config-lts/linux-intel-lts2022/virt_diffconfig:37]: let's follow same format as rest of the file 
"# CONFIG_VIRTIO_IVSHMEM_DEBUG is not set"
"# CONFIG_VIRTIO_MEM is not set"

File: groups/kernel/gmin64/config-lts/linux-intel-lts2022/x86_64_defconfig [hunk 5]
```diff
@@ -7748,6 +7747,25 @@ CONFIG_ARCH_USE_MEMTEST=y
...
+CONFIG_FS_DAX=n
+CONFIG_MEMORY_HOTPLUG_DEFAULT_ONLINE=n
+CONFIG_PCI_P2PDMA=n
+CONFIG_ARCH_ENABLE_MEMORY_HOTREMOVE=y
+CONFIG_DEVICE_MIGRATION=y
+CONFIG_GET_FREE_REGION=y
+CONFIG_HAVE_BOOTMEM_INFO_NODE=y
+CONFIG_MHP_MEMMAP_ON_MEMORY=y
+CONFIG_ZONE_DEVICE=y
+CONFIG_TEST_HMM=n
```
Comment (JeevakaPrabu) [inline groups/kernel/gmin64/config-lts/linux-intel-lts2022/x86_64_defconfig:7764]: let's follow same format as rest of the file 
"# CONFIG_TEST_HMM is not set"
"# CONFIG_FS_DAX is not set"

**Review by JeevakaPrabu — CHANGES_REQUESTED** [review]
in commit message there is a typo "andr" => "and"
Please add Tests done in commit message

File: groups/kernel/gmin64/config-lts/linux-intel-lts2022/virt_diffconfig [hunk 6]
```diff
@@ -30,3 +30,12 @@ CONFIG_GPIO_VIRTIO=y
...
+CONFIG_IVSHM=y
+CONFIG_IVSHM_IVSHMEM=y
+CONFIG_IVSHM_GUEST_SHM=y
+CONFIG_QNX_GUEST=y
```
Comment (ceiba1985) [inline groups/kernel/gmin64/config-lts/linux-intel-lts2022/virt_diffconfig:41]: Would better add a new line below the last line. Or .config cannot be generated successfully.

File: groups/kernel/gmin64/config-lts/linux-intel-lts2022/virt_diffconfig (hunk 4 above)
Comment (YuanyuanZhao609) [reply groups/kernel/gmin64/config-lts/linux-intel-lts2022/virt_diffconfig]: OK.

File: groups/kernel/gmin64/config-lts/linux-intel-lts2022/x86_64_defconfig (hunk 5 above)
Comment (YuanyuanZhao609) [reply groups/kernel/gmin64/config-lts/linux-intel-lts2022/x86_64_defconfig]: Changed format.

File: groups/kernel/gmin64/config-lts/linux-intel-lts2022/virt_diffconfig (hunk 6 above)
Comment (YuanyuanZhao609) [reply groups/kernel/gmin64/config-lts/linux-intel-lts2022/virt_diffconfig]: Thank you. Add new line to config files.

File: groups/kernel/gmin64/config-lts/linux-intel-lts2022/qnx_diffconfig [hunk 7]
```diff
@@ -0,0 +1,4 @@
+CONFIG_QNX_GUEST=y
+CONFIG_IVSHM_GUEST_SHM=y
+CONFIG_VIRTIO_IVSHMEM_GUEST_SHM=y
+
```
Comment (xyzhao2018) [inline groups/kernel/gmin64/config-lts/linux-intel-lts2022/qnx_diffconfig:4]: Please remove the space line here.

**Review by xyzhao2018 — COMMENTED** [review]
Please remove the space line here.

File: groups/kernel/gmin64/config-lts/linux-intel-lts2022/virt_diffconfig [hunk 8]
```diff
@@ -30,3 +30,10 @@ CONFIG_GPIO_VIRTIO=y
 CONFIG_I2C_VIRTIO=y
 CONFIG_ACRN_GUEST=y
 CONFIG_SPI_VIRTIO=y
+CONFIG_VIRTIO_IVSHMEM=y
+CONFIG_VIRTIO_IVSHMEM_IVSHMEM=y
+# CONFIG_VIRTIO_IVSHMEM_DEBUG is not set
+# CONFIG_VIRTIO_MEM is not set
+CONFIG_IVSHM=y
+CONFIG_IVSHM_IVSHMEM=y
+
```
Comment (xyzhao2018) [inline groups/kernel/gmin64/config-lts/linux-intel-lts2022/virt_diffconfig:39]: Please remove the space line here.

**Review by xyzhao2018 — COMMENTED** [review]
Please remove the space line here.

**Review by xyzhao2018 — APPROVED** [review]

**Review by ceiba1985 — APPROVED** [review]

**Review by bing8deng — APPROVED** [review]

**Review by bhe4 — APPROVED** [review]

**Comment by YuanyuanZhao609** [comment]
EB was tested, QNX and android works normally.

**Review by JeevakaPrabu — APPROVED** [review]

**Review by zhy1030 — APPROVED** [review]

**Pull request merged**